File: docs-examples/example-java/src/test/java/io/micronaut/ignite/docs/transaction/SampleIgniteTransaction.java
package io.micronaut.ignite.docs.transaction;

import io.micronaut.ignite.annotation.IgniteTransaction;
import org.apache.ignite.Ignite;
import org.apache.ignite.IgniteCache;

public class SampleIgniteTransaction {
    private final Ignite ignite;

    SampleIgniteTransaction(Ignite ignite){
        this.ignite = ignite;
    }

    @IgniteTransaction
    public void method() {
        IgniteCache cache = this.ignite.cache("sample");
        cache.put("key", "value");
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/ignite/docs/transaction/SampleIgniteTransaction.kt
package io.micronaut.ignite.docs.transaction

import io.micronaut.ignite.annotation.IgniteTransaction
import org.apache.ignite.Ignite
import org.apache.ignite.IgniteCache

class SampleIgniteTransaction constructor(private val ignite: Ignite) {

    @IgniteTransaction
    fun method() {
        val cache: IgniteCache<String, String> = ignite.cache("sample")
        cache.put("key", "value")
    }
}
